core/trino-main/src/main/java/io/trino/util/JsonUtil.java

```diff
@@ -124,7 +127,13 @@ public static JsonParser createJsonParser(JsonFactory factory, Slice json)
             throws IOException
     {
         // Jackson tries to detect the character encoding automatically when using InputStream
-        // so we pass an InputStreamReader instead.
+        // so we pass StringReader or an InputStreamReader instead.
+        byte[] jsonBytes = json.getBytes();
```

Comment:
move under `if` condition. USe `io.airlift.slice.Slice#length` for `STRING_READER_BYTE_ARRAY_LENGTH_LIMIT` comparison